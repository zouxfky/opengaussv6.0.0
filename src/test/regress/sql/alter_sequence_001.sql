DROP SCHEMA IF EXISTS test_alter_seq_sche_01 CASCADE;

CREATE SCHEMA test_alter_seq_sche_01;

SET CURRENT_SCHEMA TO test_alter_seq_sche_01;

create sequence seqa;

select nextval('seqa');

alter sequence seqa increment 10;

select nextval('seqa');

select nextval('seqa');

alter sequence seqa restart 200;

select nextval('seqa');

select nextval('seqa');

alter sequence seqa cycle;

alter sequence seqa start 10;

alter sequence seqa maxvalue 1000;

alter sequence seqa minvalue 5;

select sequence_name , last_value , start_value , increment_by , max_value , min_value , cache_value , is_cycled , is_called from seqa;

DROP SCHEMA test_alter_seq_sche_01 CASCADE;
