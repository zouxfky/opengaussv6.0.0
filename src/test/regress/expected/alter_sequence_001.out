DROP SCHEMA IF EXISTS test_alter_seq_sche_01 CASCADE;
NOTICE:  schema "test_alter_seq_sche_01" does not exist, skipping
CREATE SCHEMA test_alter_seq_sche_01;
SET CURRENT_SCHEMA TO test_alter_seq_sche_01;
create sequence seqa;
select nextval('seqa');
 nextval 
---------
       1
(1 row)

alter sequence seqa increment 10;
select nextval('seqa');
 nextval 
---------
      11
(1 row)

select nextval('seqa');
 nextval 
---------
      21
(1 row)

alter sequence seqa restart 200;
select nextval('seqa');
 nextval 
---------
     200
(1 row)

select nextval('seqa');
 nextval 
---------
     210
(1 row)

alter sequence seqa cycle;
alter sequence seqa start 10;
alter sequence seqa maxvalue 1000;
alter sequence seqa minvalue 5;
select sequence_name , last_value , start_value , increment_by , max_value , min_value , cache_value , is_cycled , is_called from seqa;
 sequence_name | last_value | start_value | increment_by | max_value | min_value | cache_value | is_cycled | is_called 
---------------+------------+-------------+--------------+-----------+-----------+-------------+-----------+-----------
 seqa          |        210 |          10 |           10 |      1000 |         5 |           1 | t         | t
(1 row)

DROP SCHEMA test_alter_seq_sche_01 CASCADE;
NOTICE:  drop cascades to sequence seqa
